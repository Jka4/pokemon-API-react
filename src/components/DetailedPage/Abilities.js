import React from "react";

const Abilities = (props) => {
	const abilities = props?.abilities;

	return (
		<div className='abilities'>
			<ul className='abilitiesUL'>
				<div className='skills_title'>Abilities</div>

				{abilities && abilities.map((i, key) => (
					<li className='abilitiesLI' key={key}>
						{key + 1}: {i.ability.name}
					</li>
				))}
			</ul>
		</div>
	);
};

export default Abilities;