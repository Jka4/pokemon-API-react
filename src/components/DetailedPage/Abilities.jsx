import React from "react";
import Paper from "@material-ui/core/Paper";

const Abilities = ({ abilities = [] }) => {
  return (
    <div className="abilities">
      <Paper elevation={3} className="abilitiesUL">
        <div className="skills_title">Abilities</div>

        {abilities.map((element, key) => (
          <li className="abilitiesLI" key={key}>
            {key + 1}: {element.ability.name}
          </li>
        ))}
      </Paper>
    </div>
  );
};

export default Abilities;
