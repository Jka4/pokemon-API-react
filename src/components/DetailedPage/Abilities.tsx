import React from "react";
import Paper from "@material-ui/core/Paper";
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

type Props = {
  abilities: AbilityElements[];
};

type AbilityElements = {
  ability: {
    url: string;
    name: string;
  };
}

const Abilities: React.FC<Props> = ({ abilities = [] }: Props) => {

  const fallbackSkeletons = () => {
    return (
      <SkeletonTheme color="#53aeff" highlightColor="#0066be">
        {[1, 2, 3, 4, 5, 6].map(el => (
          <p key={el} >
            <Skeleton height={15} width={100} />
          </p>
        ))}
      </SkeletonTheme>
    );
  };

  return (
    <div className="abilities">
      <Paper elevation={3} className="abilitiesUL">
        <div className="skills_title">Abilities</div>
        {abilities.length !== 0 ? (abilities.map((element: AbilityElements) => (
          <li className="abilitiesLI" key={element.ability.name}>
            - {element.ability.name}
          </li>
        ))) : (
            fallbackSkeletons()
          )}
      </Paper>
    </div>
  );
};

export default Abilities;
