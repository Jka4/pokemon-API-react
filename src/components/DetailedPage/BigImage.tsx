import React, { lazy } from 'react';
import Paper from '@material-ui/core/Paper';

import { Pokes } from 'types';

const ImageContainer = lazy(() => import('components/ImageContainer/ImageContainer'));

interface RootType {
  bigImage?: Pokes;
}

const BigImage: React.FC<RootType> = ({ bigImage }: RootType) => {
  const placeholder = () => {
    return <img src={bigImage?.placeholderBase64} className="bigImage " alt="" style={{ filter: 'blur(18px)' }} />;
  };

  return (
    <Paper elevation={3} style={{ overflow: 'hidden' }}>
      <ImageContainer url={bigImage?.imageHQ || ''} cn={'bigImage'} fallback={placeholder} />
    </Paper>
  );
};

export default BigImage;
