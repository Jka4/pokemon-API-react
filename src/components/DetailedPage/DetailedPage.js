import React, { lazy, Suspense, useEffect, useState } from "react";
import { Provider, connect } from 'react-redux';
import Loader from 'react-loader-spinner';

import Stats from '@Stats';
import Abilities from '@Abilities';
import EvolutionForms from '@EvolutionForms';

import pokemonDataArray from '@pokemonDataArray';

import store from "@Store";
import * as _ from 'lodash';

import './styles/style.scss';

const ImageContainer = lazy(() => import('@ImageContainer'));


const DetailedPage = props => {
  const data = props.detailsPageTest || props.detailsPage;
  const { abilities, stats, weight } = props.detailsPage;
  const { sprites, name } = data;
  const [evolutionChain, setEvolutionChain] = useState();

  useEffect(() => {
    let pokemonObj = _.find(pokemonDataArray, (o) => o.name === data.name);
    pokemonObj && setEvolutionChain(pokemonObj);
  }, [data])


  return (
    <React.Fragment>
      <div className='detailedPage' data-testid="detailedPageTest">
        <div className='name'>{name}</div>
        <div className='imagesLine'>
          {sprites && Object.keys(sprites).map(
            spriteName =>
              sprites[spriteName] && (
                <div className='block' key={spriteName}>
                  <Suspense fallback={<Loader type="TailSpin" height={50}
                    width={50} color={"red"}
                  />}>
                    <ImageContainer url={sprites[spriteName]} />
                  </Suspense>
                  <div className='title'>{spriteName}</div>
                </div>
              )
          )}
        </div>
        <div className="mainInformations">
          <div className='skills'>
            <Stats weight={weight} stats={stats} />
            <Abilities abilities={abilities} />
          </div>
          <div className="bigImage">
            {evolutionChain?.imageHQ && <Suspense fallback={<Loader type="TailSpin" height={320}
              width={320} color={"red"}
            />}>
              <ImageContainer url={evolutionChain.imageHQ} cn={'bigImage'} />
            </Suspense>}
          </div>
        </div>
        <EvolutionForms evolutionChain={evolutionChain} currentPokemon={name} />
      </div>
    </React.Fragment>
  );
}


const ConnectedDetailedPage = connect(store => {
  return {
    detailsPage: store.detailsPage,
  };
})(DetailedPage);

export default props => (
  <Provider store={store}>
    <ConnectedDetailedPage {...props} />
  </Provider>
);
