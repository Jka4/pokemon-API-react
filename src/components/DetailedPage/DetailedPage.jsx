import React, { lazy, Suspense, useEffect, useState } from "react";
import { Provider, connect } from "react-redux";
import Loader from "react-loader-spinner";
import PropTypes from "prop-types";

import Stats from "@Stats";
import Abilities from "@Abilities";
import EvolutionForms from "@EvolutionForms";
import Paper from "@material-ui/core/Paper";

import store from "@Store";
import find from "lodash.find";

import "./styles/style.scss";

const ImageContainer = lazy(() => import("@ImageContainer"));

const DetailedPage = (props) => {
  const data = props.detailsPageTest || props.detailsPage;
  const { sprites, name } = data;
  const {
    detailsPage: { abilities, stats, weight },
    pokemonsArr,
  } = props;
  const [evolutionChain, setEvolutionChain] = useState();

  useEffect(() => {
    let pokemonObj = find(pokemonsArr, (o) => o.name === data.name);
    pokemonObj && setEvolutionChain(pokemonObj);
  }, [data, pokemonsArr]);

  return (
    <>
      <div className="detailedPage" data-testid="detailedPageTest">
        <div className="name">{name}</div>
        <div className="imagesLine">
          {sprites &&
            Object.keys(sprites).map(
              (spriteName) =>
                sprites[spriteName] && (
                  <Paper elevation={3} className="block" key={spriteName}>
                    <Suspense
                      fallback={
                        <Loader
                          type="TailSpin"
                          height={50}
                          width={50}
                          color={"red"}
                        />
                      }
                    >
                      <ImageContainer url={sprites[spriteName]} />
                    </Suspense>
                    <div className="title">{spriteName}</div>
                  </Paper>
                )
            )}
        </div>
        <div className="mainInformations">
          <div className="skills">
            <Stats weight={weight} stats={stats} />
            <Abilities abilities={abilities} />
          </div>

          {evolutionChain?.imageHQ && (
            <Paper elevation={3} className="bigImage">
              {evolutionChain?.imageHQ && (
                <Suspense
                  fallback={
                    <img
                      src={evolutionChain?.placeholderBase64}
                      className="placeholderBase64 bigImage"
                      alt=""
                    />
                  }
                >
                  <ImageContainer
                    url={evolutionChain.imageHQ}
                    cn={"bigImage"}
                  />
                </Suspense>
              )}
            </Paper>
          )}
        </div>

        <EvolutionForms
          evolutionChain={evolutionChain}
          currentPokemon={name}
          pokemonsArr={pokemonsArr}
        />
      </div>
    </>
  );
};

const ConnectedDetailedPage = connect((store) => {
  return {
    detailsPage: store.detailsPage,
    pokemonsArr: store.pokemonsArr,
  };
})(DetailedPage);

export default (props) => (
  <Provider store={store}>
    <ConnectedDetailedPage {...props} />
  </Provider>
);

DetailedPage.propTypes = {
  detailsPage: PropTypes.object,
  pokemonsArr: PropTypes.array,
};
