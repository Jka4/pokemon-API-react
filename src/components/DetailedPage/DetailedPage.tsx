/* eslint-disable import/no-anonymous-default-export */
import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

import { Provider, connect } from 'react-redux';
import store from 'Store/store';

import Stats from './Stats';
import Abilities from './Abilities';
import Sprites from './Sprites';
import EvolutionForms from './EvolutionForms';
import BigImage from './BigImage';

import axios from 'axios';

import './styles/style.scss';

import { Pokes, IStoreType } from 'types/index';

interface DetailedPageType {
  pokemonArr: Pokes[];
}

const DetailedPage: React.FC<DetailedPageType> = ({ pokemonArr = [] }: DetailedPageType) => {
  const [bigImage, setBigImage] = useState<Pokes>();
  const [detailedPage, setDetailedPage] = useState<any>({});

  const { abilities, stats, weight, sprites } = detailedPage;

  const currentPokemon = useLocation().pathname.split('/').pop();

  useEffect(() => {
    setDetailedPage({});
    let isSubscribed: boolean = true;

    const URL = `https://pokeapi.co/api/v2/pokemon/${currentPokemon}/`;
    axios.get(URL).then(async (response) => {
      const poke = response.data;

      delete poke.sprites.other;
      delete poke.sprites.versions;

      isSubscribed && setDetailedPage(poke);
    });

    return (): void => {
      isSubscribed = false;
    };
  }, [currentPokemon]);

  useEffect(() => {
    const pokemon: Pokes | undefined = pokemonArr.find((el) => el.name === currentPokemon);
    pokemon && setBigImage(pokemon);
  }, [currentPokemon, pokemonArr]);

  return (
    <>
      <div className="detailedPage">
        <div className="name">{currentPokemon || 'POKEMON'}</div>

        <EvolutionForms currentPokemon={currentPokemon} pokemonArr={pokemonArr} />
        <div className="mainInformation">
          <div className="skills">
            <Stats weight={weight} stats={stats} />
            <Abilities abilities={abilities} />
          </div>
          <BigImage bigImage={bigImage} />
        </div>

        <Sprites sprites={sprites} />
      </div>
    </>
  );
};

// export default DetailedPage;

const ConnectedDetailedPage = connect((store: IStoreType) => {
  return {
    pokemonArr: store.pokemonArr,
  };
})(DetailedPage);

export default (props = {}) => (
  <Provider store={store}>
    <ConnectedDetailedPage {...props} />
  </Provider>
);
