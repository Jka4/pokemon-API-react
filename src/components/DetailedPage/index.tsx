/* eslint-disable import/no-anonymous-default-export */
import React, { useEffect, useState } from 'react';

import { useDispatch, useSelector } from 'react-redux';
import { useRouteMatch } from 'react-router-dom';

import Stats from './views/Stats';
import Abilities from './views/Abilities';
import Sprites from './views/Sprites';
import EvolutionForms from './views/EvolutionForms';
import BigImage from './views/BigImage';

import { PokesTypes } from 'types/index';

import styled from 'styled-components';
import { detailedPageSelector, pokemonArrSelector } from 'selectors';
import getDetailedPokemon from 'ac/getDetailedPokemon';

interface DetailedPageType {
  pokemonArr: PokesTypes[];
  detailedPage: PokesTypes;
}

const DetailedPage: React.FC = () => {
  const {
    params: { pokemonId: currentName },
  } = useRouteMatch<any>();

  const dispatch = useDispatch();

  const pokemonArr = useSelector(pokemonArrSelector);
  const detailedPage = useSelector(detailedPageSelector);

  const [bigImageProps, setBigImageProps] = useState<any>({});

  const { abilities = [], stats = [], weight = 0, sprites = {} } = detailedPage;
  const { imageHQ = '', placeholderBase64 = '' } = bigImageProps;

  useEffect(() => {
    dispatch(getDetailedPokemon(currentName));
  }, [currentName, pokemonArr]);

  useEffect(() => {
    const pokemon: any = pokemonArr.find((el: any) => el.name === currentName);
    setBigImageProps(pokemon);
  }, [currentName, pokemonArr]);

  return (
    <>
      <DetailedPageWrapper>
        <Name>{currentName || '...'}</Name>

        <EvolutionForms currentPokemon={currentName} pokemonArr={pokemonArr} />

        <MainInformation>
          <Skills>
            <Stats weight={weight} stats={stats} />
            <Abilities abilities={abilities} />
          </Skills>
          <BigImage imageHQ={imageHQ} placeholderBase64={placeholderBase64} />
        </MainInformation>

        <Sprites sprites={sprites} />
      </DetailedPageWrapper>
    </>
  );
};

const DetailedPageWrapper = styled.div`
  margin-top: 100px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  height: auto;
`;

const Name = styled.div`
  width: 100%;
  height: auto;
  text-align: center;
  font-size: 4rem;

  &::first-letter {
    text-transform: uppercase;
  }
`;

const MainInformation = styled.div`
  min-height: 300px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: flex-start;
  margin-top: 20px;
  margin-bottom: 100px;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    flex-direction: column;
    width: 100%;
    padding: 20px;
    margin-bottom: 10px;
  }
`;

const Skills = styled.div`
  display: flex;
  flex-direction: row;
  margin-right: 30px;

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    order: 2;
    margin-top: 20px;
    width: 100%;
  }

  height: auto;
`;

export default DetailedPage;
