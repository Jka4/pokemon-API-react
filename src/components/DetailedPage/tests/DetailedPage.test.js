import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { setDelailedPageData } from "@APIutils";
import DetailedPage from '@DetailedPage';

test('Fetch detailed page', async () => {
	await setDelailedPageData(1);
	render(
		<Router>
			<DetailedPage />
		</Router>);

	const detailedPage = await screen.findByTestId('detailedPageTest');

	expect(detailedPage).toHaveTextContent('bulbasaur');
	expect(detailedPage).toHaveTextContent('Abilities');
	expect(detailedPage).toHaveTextContent('Stats');
	expect(detailedPage).toHaveTextContent('front_default');
	expect(detailedPage).toHaveTextContent('back_default');
})
