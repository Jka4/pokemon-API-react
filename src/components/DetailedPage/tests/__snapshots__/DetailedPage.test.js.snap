// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedPage DetailedPage renders correctly 1`] = `
<div
  className="detailedPage"
  data-testid="detailedPageTest"
>
  <div
    className="name"
  >
    bulbasaur
  </div>
  <div
    className="imagesLine"
  >
    <div
      className="MuiPaper-root block MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          height={50}
          viewBox="0 0 38 38"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stopColor="red"
                stopOpacity="0"
              />
              <stop
                offset="63.146%"
                stopColor="red"
                stopOpacity=".631"
              />
              <stop
                offset="100%"
                stopColor="red"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="red"
                strokeWidth="2"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r={1}
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      className="MuiPaper-root block MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          height={50}
          viewBox="0 0 38 38"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stopColor="red"
                stopOpacity="0"
              />
              <stop
                offset="63.146%"
                stopColor="red"
                stopOpacity=".631"
              />
              <stop
                offset="100%"
                stopColor="red"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="red"
                strokeWidth="2"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r={1}
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      className="MuiPaper-root block MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          height={50}
          viewBox="0 0 38 38"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stopColor="red"
                stopOpacity="0"
              />
              <stop
                offset="63.146%"
                stopColor="red"
                stopOpacity=".631"
              />
              <stop
                offset="100%"
                stopColor="red"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="red"
                strokeWidth="2"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r={1}
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      className="MuiPaper-root block MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          height={50}
          viewBox="0 0 38 38"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stopColor="red"
                stopOpacity="0"
              />
              <stop
                offset="63.146%"
                stopColor="red"
                stopOpacity=".631"
              />
              <stop
                offset="100%"
                stopColor="red"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="red"
                strokeWidth="2"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r={1}
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="mainInformations"
  >
    <div
      className="skills"
    >
      <div
        className="stats"
      >
        <div
          className="MuiPaper-root statsUL MuiPaper-elevation3 MuiPaper-rounded"
        >
          <div
            className="skills_title"
          >
            Stats
          </div>
          <li
            className="statsLI"
          >
            hp
             
            45
          </li>
          <li
            className="statsLI"
          >
            attack
             
            49
          </li>
          <li
            className="statsLI"
          >
            defense
             
            49
          </li>
          <li
            className="statsLI"
          >
            special-attack
             
            65
          </li>
          <li
            className="statsLI"
          >
            special-defense
             
            65
          </li>
          <li
            className="statsLI"
          >
            speed
             
            45
          </li>
          <li
            className="statsLI"
          >
            weight 
            69
            kg
          </li>
        </div>
      </div>
      <div
        className="abilities"
      >
        <div
          className="MuiPaper-root abilitiesUL MuiPaper-elevation3 MuiPaper-rounded"
        >
          <div
            className="skills_title"
          >
            Abilities
          </div>
          <li
            className="abilitiesLI"
          >
            1
            : 
            overgrow
          </li>
          <li
            className="abilitiesLI"
          >
            2
            : 
            chlorophyll
          </li>
        </div>
      </div>
    </div>
  </div>
  <div
    className="formsTitle"
  >
    <span>
      Form:
    </span>
  </div>
  <div
    className="evolutionForms"
  />
</div>
`;
